pyuvdata/uvdata/corr_fits_src/corr_fits.pyx

```diff
@@ -95,30 +96,35 @@ cpdef tuple generate_map(
             )
             map_inds[bls_ind * 4 + pol_ind] = data_index
 
-  return map_inds, conj
+  return
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
 cpdef numpy.ndarray[ndim=1, dtype=numpy.float64_t] get_cable_len_diffs(
   int Nblts,
-  numpy.ndarray[dtype=numpy.int_t, ndim=1] ant1_array,
-  numpy.ndarray[dtype=numpy.int_t, ndim=1] ant2_array,
+  numpy.int_t[:] ant1_array,
+  numpy.int_t[:] ant2_array,
   numpy.ndarray cable_lens,
 ):
-  cdef int i
-  cdef numpy.ndarray[ndim=1, dtype=numpy.float64_t] cable_array = np.zeros(len(cable_lens), dtype=np.float_)
-  cdef numpy.ndarray[ndim=1, dtype=numpy.float64_t] cable_diffs = np.zeros(Nblts, dtype=np.float_)
-
+  cdef Py_ssize_t i
+  cdef int n_cables = cable_lens.shape[0]
+  cdef numpy.float64_t[::1] cable_array = np.zeros(n_cables, dtype=np.float_)
+  cdef numpy.float64_t[::1] cable_diffs = np.zeros(Nblts, dtype=np.float_)
   # "the velocity factor of electic fields in RG-6 like coax"
   # from MWA_Tools/CONV2UVFITS/convutils.h
   cdef float v_factor = 1.204
 
   # check if the cable length already has the velocity factor applied
-  for i in range(len(cable_lens)):
-    if cable_lens[i][0:3] == "EL_":
+  # this loop is very python-y but there is no real gain to trying to make it C-esque
+  for i in range(n_cables):
+
+    if cable_lens[i][:3] == "EL_":
       cable_array[i] = float(cable_lens[i][3:])
     else:
       cable_array[i] = float(cable_lens[i]) * v_factor
 
   # build array of differences
-  cable_diffs = cable_array[ant2_array] - cable_array[ant1_array]
+  for i in range(Nblts):
+    cable_diffs[i] = cable_array[ant2_array[i]] - cable_array[ant1_array[i]]
 
-  return cable_diffs
+  return np.asarray(cable_diffs)
```

Comment:
Nice to cast the memory view to an array to make it more user-friendly 👍

Reply:
This isn't 100% related, but it turns out that if you compile to C and return a memoryview it often casts back to an array and everything is fine. if you compile to C++ it returns a memoryview object, so I've been trying to get in the habit of casting it explicitly when the code says it should be an array.